addons/sale_loyalty/tests/test_program_multi_company.py

```diff
@@ -81,3 +81,28 @@ def test_applicable_programs(self):
         order_b._update_programs_and_rewards()
         self.assertIn(self.immediate_promotion_program_c2, order_b._get_applied_programs())
         self.assertNotIn(self.immediate_promotion_program, order_b._get_applied_programs())
+
+    def test_applicable_programs_on_branch(self):
+        # create a branch
+        branch_a = self.env['res.company'].create(dict(name="Branch A", parent_id=self.company_a.id))
+
+        # create an order
+        order = self.empty_order
+        order.write({'order_line': [
+            (0, False, {
```

Comment:
Use `Command.create` directly, it's clearer than these magic commands =]

Reply:
Understood! I made the revisions!